riff-raff/app/deployment/actors/TasksRunner.scala

```diff
@@ -37,7 +37,7 @@ class TasksRunner(stopFlagAgent: Agent[Map[UUID, String]]) extends Actor with Lo
                   log.debug(s"Running task $taskId")
                   deployMetricsProcessor ! TaskStart(uuid, taskId, queueTime, new DateTime())
                   deployReporter.taskContext(task) { taskReporter =>
-                    task.execute(taskReporter, stopFlagAsker)
+                    task.execute(rootResources, stopFlagAsker)
```

Comment:
You're dropping the reporter context here. The context changes here on 28 and also 39 with the calls to `infoContext` and `deployContext`. This creates the nesting in the logs which I suspect may have gone missing in your test runs?

It is somewhat less elegant as you'll need to recreate the DeployResources with the new context. Can't immediately think of a neater way of solving it though, other than passing in the stsClient elsewhere again 🙄